src/xunit.analyzers/AssertEqualLiteralValueShouldBeFirst.cs

```diff
@@ -18,21 +19,31 @@ public class AssertEqualLiteralValueShouldBeFirst : AssertUsageAnalyzerBase
 
         protected override void Analyze(SyntaxNodeAnalysisContext context, InvocationExpressionSyntax invocation, IMethodSymbol method)
         {
-            if (invocation.ArgumentList.Arguments.Count < 2)
+            var arguments = invocation.ArgumentList.Arguments;
+            if (arguments.Count < 2)
                 return;
 
-            var firstArg = invocation.ArgumentList.Arguments[0];
-            var secondArg = invocation.ArgumentList.Arguments[1];
+            ArgumentSyntax expectedArg, actualArg;
+            if (arguments.All(x => x.NameColon != null))
+            {
+                expectedArg = arguments.Single(x => x.NameColon.Name.Identifier.Text == "expected");
+                actualArg = arguments.Single(x => x.NameColon.Name.Identifier.Text == "actual");
+            }
+            else
+            {
+                expectedArg = arguments[0];
+                actualArg = arguments[1];
+            }
 
-            if (IsLiteralOrConstant(secondArg.Expression, context.SemanticModel, context.CancellationToken) &&
-                !IsLiteralOrConstant(firstArg.Expression, context.SemanticModel, context.CancellationToken))
+            if (IsLiteralOrConstant(actualArg.Expression, context.SemanticModel, context.CancellationToken) &&
+                !IsLiteralOrConstant(expectedArg.Expression, context.SemanticModel, context.CancellationToken))
```

Comment:
💭 It seems this would `NullReferenceException` if `exp:` is used instead of `expected:`.

Reply:
That's the thing, I wanted to cover that, so I put in [a test](https://github.com/xunit/xunit.analyzers/pull/113/files#diff-66f2b6b7a892d46bdf7cb3cabd0a84fdR96) for that, and I just couldn't get it to fail. I also found out that this project actually already handles [AD0001](), but that didn't fire (I sanity checked it by simply forcing a NRE, when the tree is valid, and then I did got that diagnostic as a test failure).

So I wanted to take the host environment out of the equation, and checked in VS (using this [commit](https://github.com/tiesmaster/xunit.analyzers/commit/2b5e10042ce922b9c27b68349d4a115ff15524ea)), with a breakpoint on the `return` (L33), and that just doesn't fire, whatever I type.

So I thought Roslyn might _short-circuit_ analyzers in certain cases. Doesn't ring a bell? Or maybe a bug? Anyhow, I'm going to browse Roslyn to see if I can find anything.

Reply:
:arrow_right: It's handled here:

https://github.com/xunit/xunit.analyzers/blob/ad63ad341c3cc50c934a62d59da4e551daf236f9/src/xunit.analyzers/Analysis/AssertUsageAnalyzerBase.cs#L42-L44

Reply:
Hmm, that's embarrassing. It was right in front of me, all along. I'm not too familiar with this code base yet, but I _did_ take a look at this, but I guessed I just glanced over this 😕 Thanks for your feedback, and time. Next time, I'll take a better look first.